release/models/gnpsi/openconfig-gnpsi.yang

```diff
@@ -0,0 +1,156 @@
+module openconfig-gnpsi {
+  yang-version "1";
+  namespace "http://openconfig.net/yang/gnpsi";
+  prefix "oc-gnpsi";
+
+  import openconfig-extensions {
+    prefix oc-ext;
+  }
+
+  import openconfig-system {
+    prefix oc-sys;
+  }
+
+  import openconfig-system-grpc {
+    prefix oc-sys-grpc;
+  }
+
+  import openconfig-gnpsi-types {
+    prefix oc-gnpsit;
+  }
+
+  import openconfig-inet-types {
+    prefix oc-inet;
+  }
+
+  import openconfig-yang-types {
+    prefix oc-yang;
+  }
+
+  organization
+    "OpenConfig Working group";
+  contact
+    "www.openconfig.net";
+
+  description
+    "This module defines configuration and operational state data
+    related to gRPC Network Packet Sampling Interface (gNPSI).
+
+    GNPSI Proto Reference: https://github.com/openconfig/gnpsi";
+
+  oc-ext:openconfig-version "0.1.0";
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  revision 2024-05-16 {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  grouping gnpsi-base {
+    description
+      "Top-level gNPSI base container.";
+    uses connections-top;
+  }
+
+  grouping grpc-server-connections-state {
+    description
+      "Operational data for gNPSI connections.";
+
+    leaf address {
+      type oc-inet:ip-address;
+      description
+        "IPv4/IPv6 address of the gNPSI connection.";
+    }
+
+    leaf port {
+      type oc-inet:port-number;
+      description
+        "UDP port number for the gNPSI connection.";
+    }
+  }
+
+  grouping connections-top {
```

Comment:
There are no gNPSI connections; there are only gRPC connections with gNPSI streams (clients) running on top of them. If, as @dplore suggested, the intention is to expose grpc/tcp state information, then all of this should be modeled as generic gRPC state outside of gNPSI model.

Reply:
Revised. Connections are in generic gRPC state now :)